### DISCUSSION TITLE
Need more info on using New-AWSPowerShellLambdaPackage

### DISCUSSION THREAD

**Discussion by halciber** [aws/aws-lambda-dotnet#1766] [Q&A] [answered]
### Describe the issue

When using New-AWSPowerShellLambdaPackage, I'm trying to specify the specific .NET Core SDK to use, but the information in the cmdlet's documentation isn't quite enough. Also, there isn't an example using the -PowerShellSdkVersion. 

I've put in several values for this parameter, but none of it is working. Finally the documentation calls for using the "Microsoft.PowerShell.SDK NuGet package", but I'm not sure how to determine this.

Any help would be welcome.

Thanks,
Mike Goldweber

### Links

https://github.com/aws/aws-lambda-dotnet/blob/master/PowerShell/Module/Public/New-AWSPowerShellLambdaPackage.ps1

**Comment by ashishdhingra** [comment, marked as answer]
@halciber Good morning. Thanks for the question. Please refer to blog post [Deploying PowerShell-based Lambda with AWS CloudFormation](https://aws.amazon.com/blogs/developer/deploying-powershell-based-lambda-with-aws-cloudformation/) on how to use CmdLet `New-AWSPowerShellLambdaPackage`. The [$PowerShellSdkVersion](https://github.com/aws/aws-lambda-dotnet/blob/f6157b550b7f83ede4e0546d58237ef5c5549580/PowerShell/Module/Public/New-AWSPowerShellLambdaPackage.ps1#L76C17-L76C37) per documentation [here](https://github.com/aws/aws-lambda-dotnet/blob/f6157b550b7f83ede4e0546d58237ef5c5549580/PowerShell/Module/Public/New-AWSPowerShellLambdaPackage.ps1#L32) is an `Optional parameter to override the version of PowerShell that will execute the script. The version number  must match a version of the Microsoft.PowerShell.SDK NuGet package. https://www.nuget.org/packages/Microsoft.PowerShell.SDK`. This is used in helper function [_addPowerShellLambdaProjectContent](https://github.com/aws/aws-lambda-dotnet/blob/f6157b550b7f83ede4e0546d58237ef5c5549580/PowerShell/Module/Private/_ProjectCreationFunctions.ps1#L100C10-L100C44) and would default to value defined [here](https://github.com/aws/aws-lambda-dotnet/blob/f6157b550b7f83ede4e0546d58237ef5c5549580/PowerShell/Module/Private/_Constants.ps1#L23C3-L23C4), if missing.

Thanks,
Ashish

**Reply by halciber** [reply]
Hi Ashish,

Thank you for this information. Respectfully, I do understand what you have written. My problem is correctly identifying the Microsoft.PowerShell.SDK Nuget package number, as well as how to put this information into the cmdlet's parameter.

The documentation does not show an example of this. The version number I have doesn't work. So, I'm left trying to figure how where this version number can be found on my PC.

Thanks again for your efforts to help me.

Sincerely,
Mike

**Reply by ashishdhingra** [reply]
@halciber Thanks for your reply. You could get detailed description of all supported parameters using standard `Get-Help New-AWSPowerShellLambdaPackage -Detailed` PowerShell command.

Thanks,
Ashish

**Reply by halciber** [reply]
Hi Ashish,

Thanks for trying to help. I tried the -detailed parameter. Admittedly, this is new, but this provides the same information as the -examples parameter. However, in either case there is not an example of the -PowerShellSdkVersion.

More importantly, the documentation indicates that the Nuget version number is needed. I am looking at how to correctly determine this. When I look at different parts of my .NET CORE installed, I've seen two possible places for the version number:

![image](https://github.com/aws/aws-lambda-dotnet/assets/5295146/2fd0040d-ab1f-4f7c-828d-52edb8af3b2f)

and 

![image](https://github.com/aws/aws-lambda-dotnet/assets/5295146/63eb9ccf-0bf4-4ba7-b38e-19a0f7ca4884)

I've tried both "6.0.27" and the "6.0.419" without luck. So, I'm either entering in these numbers incorrectly, or I'm using the wrong number. The way the documentation is written, I can't tell which way to go from here.

Thanks for your help,
Mike
